react/utils/normalize.ts

```diff
@@ -67,6 +76,96 @@ export function changeImageUrlSize(
   return `${normalizedImageUrl}${queryStringSeparator}width=${width}&height=${height}&aspect=true`
 }
 
+function getPriceBySeller(seller: { commertialOffer: { Price: number } }) {
+  return seller.commertialOffer.Price
+}
+
+/**
+ * @description
+ * Find the lowest/highest `Price` between sellers
+ * @param item an SKU
+ * @param condition the condition enum used for comparison functions
+ * */
+function getPriceByCondition(item: { sellers: any }, condition: ConditionRule) {
+  // First, if there's only 1 seller, avoid filtering
+  const { sellers } = item
+
+  if (sellers.length === 1) return sellers[0].commertialOffer.Price
+
+  const availableSellers = sellers.filter(
+    ({ commertialOffer }: any) =>
+      commertialOffer.AvailableQuantity != null &&
+      commertialOffer.AvailableQuantity > 0
+  )
+
+  let itemPrice
+
+  if (condition === 'expensive') {
+    itemPrice = Math.max(...availableSellers.map(getPriceBySeller))
+  } else {
+    itemPrice = Math.min(...availableSellers.map(getPriceBySeller))
+  }
+
+  return itemPrice
+}
+
+/**
+ * @description
+ * Retrieves the SKU with the best price considering a provided condition
+ * @param items an array of SKUs
+ * @param condition the condition enum used for comparison functions
+ * */
+function getBestSKUPrice(items: any[], condition: ConditionRule) {
+  // First, if none or only 1 sku is available, avoid reducing
+  const filteredItems = items.filter(getOnlyAvailable)
+  if (filteredItems.length === 0) return items[0]
+  if (filteredItems.length === 1) return filteredItems[0]
+
+  return filteredItems.reduce((acc: any, curr: any) => {
+    if (condition === 'expensive') {
+      return getPriceByCondition(curr, condition) >
+        getPriceByCondition(acc, condition)
+        ? curr
+        : acc
+    }
+
+    return getPriceByCondition(curr, condition) <
+      getPriceByCondition(acc, condition)
+      ? curr
+      : acc
+  })
+}
+
+/**
+ * @description
+ * This will retrieve the specific SKU using a custom Product (field)
+ * in the Catalog. The specification has the value of the SKU ID
+ * that will be looking for
+ * @see {@link https://vtex.io/docs/recipes/all} recipe on how to implement this
+ * */
+function getSpecificSKU(items: any[], specifications: any[]) {
+  const defaultSKUspec =
+    specifications.find(
+      (spec: { name: string }) => spec.name === 'DefaultSKUSelected'
+    ) ?? null
+
+  if (!defaultSKUspec) return items[0]
```

Comment:
I'm really not sure about this. Imagine the scenario: the client wants some products to follow their default sku specification, and the rest to be ordered following `PRICE_ASC`. It wouldn't be possible with this code, since if products don't have a default SKU specified, the selected SKU will always be the first one, and there'll be no way of changing that.

What I propose is: `getSpecificSKU` or `findSKUBySpecification` (however you name it) should be the default choosing option, and should always apply if such a specification is set for that product. It shouldn't be a choice on the theme. If the client added the specification to the product, that's what they want, right?

That wouldn't change the behavior of current stores, since they don't have any product with that specification set. This way, after not finding any SKU with that specification, the code would use the specified `PreferredSKU` option, which by default would be `FIRST_AVAILABLE`, thus not changing how the store currently works. What do you think?

Reply:
A really good idea @icazevedo!
I made it so it first checks if the property exists, and if so attempts to find the desire SKU ID.
If that fails, it will default to the condition supplied.